pkg/controller/util.go

```diff
@@ -170,6 +171,17 @@ func newScratchPersistentVolumeClaimSpec(pvc *corev1.PersistentVolumeClaim, pod
 // createScratchPersistentVolumeClaim creates and returns a pointer to a scratch PVC which is created based on the passed-in pvc and storage class name.
 func createScratchPersistentVolumeClaim(client client.Client, pvc *corev1.PersistentVolumeClaim, pod *corev1.Pod, name, storageClassName string, installerLabels map[string]string, recorder record.EventRecorder) (*corev1.PersistentVolumeClaim, error) {
 	scratchPvcSpec := newScratchPersistentVolumeClaimSpec(pvc, pod, name, storageClassName)
+
+	sizeRequest := scratchPvcSpec.Spec.Resources.Requests[corev1.ResourceStorage]
+	fsOverhead, err := GetFilesystemOverhead(context.TODO(), client, scratchPvcSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get filesystem overhead for scratch PVC")
+	}
+	fsOverheadFloat, _ := strconv.ParseFloat(string(fsOverhead), 64)
+	usableSpaceRaw := util.CalculateOverheadSpace(fsOverheadFloat, sizeRequest.Value())
+
+	scratchPvcSpec.Spec.Resources.Requests[corev1.ResourceStorage] = *resource.NewScaledQuantity(usableSpaceRaw, 0)
+
```

Comment:
Correct me if I'm wrong but for FileSystem access mode if target storageclass and scratch storagclass are the same you should be safe to use the same size as target?

Reply:
Right, but there is nothing requiring this, I need to take a look at this again, I was originally only doing this if the target was block, but it is possible scratch and target are both filesystem but different overhead. So I need to calculate the size we are targeting in the target, and then based on that calculate the appriopriate scratch space size.